File: Game_Of_Life_Python/Organisms/Plants/grass.py
from Organisms.organism import OrganismInitialData
from Organisms.plant import Plant
import random
from Organisms.animal import CollisionTypes

class Grass(Plant):

    static_counter = 0

    def __init__(self, name, row, column, given_strength = -1, given_initiative = -1):
        self.strength = OrganismInitialData[name]["strength"]
        self.initiative = OrganismInitialData[name]["initiative"]
        self.character = OrganismInitialData[name]["character"]
        if given_strength != -1: self.strength = given_strength
        if given_initiative != -1: self.initiative = given_initiative
        super().__init__(self.strength, self.initiative, name, self.character, row, column, "grass.png")
        Grass.static_counter += 1
        self.type = Grass

    def action(self, grid_board):
        return self.default_plant_action(grid_board, Grass, "Grass")

    def collision(self, grid_board, organisms, current_index):
        return self.default_plant_collision()


    def get_static_counter(self):
        return Grass.static_counter

    def increase_static_counter(self):
        Grass.static_counter += 1

    def decrease_static_counter(self):
        Grass.static_counter -= 1

File: GameOfLife_Cpp/source/src/grass.cpp
#include "grass.hpp"
#include <iostream>

int Grass::GRASS_STATIC_COUNTER = 0;

Grass::Grass(int row, int column)
    : Plant(0, 0, 0, "Grass", 'G', row, column, "grass.png", OrganismType::GRASS)
{
    GRASS_STATIC_COUNTER++;
}

Grass::Grass(int row, int column, int strength, int initiative, int age)
    : Plant(strength, initiative, age, "Grass", 'G', row, column, "grass.png", OrganismType::GRASS)
{
    GRASS_STATIC_COUNTER++;
}

Grass::~Grass()
{
    GRASS_STATIC_COUNTER--;
}

ActionResult Grass::action(std::vector<std::vector<char>>& grid_board)
{
    return this->default_action_plant(grid_board);
}

CollisionResult Grass::collision(std::vector<std::vector<char>>& grid_board, std::vector<Organism*>& organisms, int current_index)
{
    return this->default_collision_plant();
}

int Grass::get_static_counter()
{
    return GRASS_STATIC_COUNTER;
}